packages/mermaid/src/diagrams/git/gitGraphAst.ts

```diff
@@ -0,0 +1,510 @@
+}
+
+/**
+ * @param list - list of items
+ * @param fn -  function to get the key
+ */
+function uniqBy(list: any[], fn: (item: any) => any) {
+  const recordMap = Object.create(null);
+  return list.reduce((out, item) => {
+    const key = fn(item);
+    if (!recordMap[key]) {
+      recordMap[key] = true;
+      out.push(item);
+    }
+    return out;
+  }, []);
+}
+
+export const setDirection = function (dir: DiagramOrientation) {
+  state.records.direction = dir;
+};
+
+export const setOptions = function (rawOptString: string) {
+  log.debug('options str', rawOptString);
+  rawOptString = rawOptString?.trim();
+  rawOptString = rawOptString || '{}';
+  try {
+    state.records.options = JSON.parse(rawOptString);
+  } catch (e: any) {
+    log.error('error while parsing gitGraph options', e.message);
+  }
+};
+
+export const getOptions = function () {
+  return state.records.options;
+};
+
+export const commit = function (msg: string, id: string, type: number, tags?: string[]) {
+  log.info('commit', msg, id, type, tags);
+  log.debug('Entering commit:', msg, id, type, tags);
+  const config = getConfig();
+  id = common.sanitizeText(id, config);
+  msg = common.sanitizeText(msg, config);
+  tags = tags?.map((tag) => common.sanitizeText(tag, config));
+  const newCommit: Commit = {
+    id: id ? id : state.records.seq + '-' + getID(),
+    message: msg,
+    seq: state.records.seq++,
+    type: type ?? commitType.NORMAL,
+    tags: tags ?? [],
+    parents: state.records.head == null ? [] : [state.records.head.id],
+    branch: state.records.currBranch,
+  };
+  state.records.head = newCommit;
+  log.info('main branch', mainBranchName);
+  state.records.commits.set(newCommit.id, newCommit);
+  state.records.branches.set(state.records.currBranch, newCommit.id);
+  log.debug('in pushCommit ' + newCommit.id);
+};
+
+export const branch = function (name: string, order?: number) {
+  name = common.sanitizeText(name, getConfig());
+  if (state.records.branches.has(name)) {
+    throw new Error(
+      `Trying to create an existing branch. (Help: Either use a new name if you want create a new branch or try using "checkout ${name}")`
+    );
+  }
+
+  state.records.branches.set(name, state.records.head != null ? state.records.head.id : null);
+  state.records.branchConfig.set(name, { name, order });
+  checkout(name);
+  log.debug('in createBranch');
+};
+
+export const merge = (
+  otherBranch: string,
+  customId?: string,
+  overrideType?: number,
+  customTags?: string[]
+): void => {
+  const config = getConfig();
+  otherBranch = common.sanitizeText(otherBranch, config);
+  if (customId) {
+    customId = common.sanitizeText(customId, config);
+  }
+  const currentBranchCheck: string | null | undefined = state.records.branches.get(
+    state.records.currBranch
+  );
+  const otherBranchCheck: string | null | undefined = state.records.branches.get(otherBranch);
+  const currentCommit: Commit | undefined = currentBranchCheck
```

Comment:
nit: Implicit types are enough. 

```suggestion
  const currentBranchCheck = state.records.branches.get(
    state.records.currBranch
  );
  const otherBranchCheck = state.records.branches.get(otherBranch);
  const currentCommit = currentBranchCheck
```